### PR TITLE
Manual Hopper Shifting

### PR DESCRIPTION
**Pull request by NikhilSuresh24**
Added a switch that enables and disables automatic hopper movement.

### CODE REVIEW

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java [hunk 1]
```diff
@@ -62,9 +66,14 @@ public void bindCommands() {
 		RobotMap.Component.driverXbox.dPad.right.whenPressed(new ChassisTurnAbsolute(RobotMap.Component.chassis, 90,
 			RobotMap.Component.navx, RobotMap.Component.chassisDriveMC));
 		RobotMap.Component.driverXbox.dPad.right.whenReleased(normalDrive);
+		RobotMap.Component.teensyStick.getButton(6).whenPressed(new SetOverride(true, new BallioOuttake()));
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: Move these down to where the normalDrive button is bound

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java [hunk 2]
```diff
@@ -62,9 +66,14 @@ public void bindCommands() {
...
+		RobotMap.Component.teensyStick.getButton(6).whenReleased(new SetOverride(false, new BallioOuttake()));
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: What about all the other places the hopper is shifted / will be shifted? Like the shooter? How do we deal with those?

File: src/org/usfirst/frc4904/robot/commands/BallioOuttake.java [hunk 3]
```diff
@@ -3,18 +3,35 @@
...
 		addSequential(new HopperSetBallio());
 		addParallel(new MotorConstant(RobotMap.Component.ballIO.directionalRoller, BallIO.DIRECTIONAL_ROLLER_OUTTAKE_SPEED));
 		addParallel(new MotorConstant(RobotMap.Component.ballIO.elevatorAndIntakeRoller,
 			BallIO.ELEVATOR_AND_INTAKE_ROLLER_FORWARD_SPEED));
 		addSequential(new MotorConstant(RobotMap.Component.ballIO.hopperRollers, BallIO.HOPPER_ROLLERS_FORWARD_SPEED));
 	}
+
+	@Override
+	public void setOverride(boolean setValue) {
+		BallioOuttake.ballioCycleOverride = setValue;
```
Comment (c0d3rman) [inline src/org/usfirst/frc4904/robot/commands/BallioOuttake.java]: Why is this static?

Comment (c0d3rman) [reply src/org/usfirst/frc4904/robot/commands/BallioOuttake.java]: Also same thing about `setValue` -> `ballioCycleOverride`

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java (hunk 2 above)
Comment (c0d3rman) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: Or BallioCycle

**Review by c0d3rman — CHANGES_REQUESTED** [review]
We need to think about how to do this better. We may need to do something similar to what we did with autoshifting.

File: src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java (hunk 2 above)
Comment (NikhilSuresh24) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: made all the commands with hopper shifts into runIfs that only run when override is false

Comment (NikhilSuresh24) [reply src/org/usfirst/frc4904/robot/humaninterface/drivers/NathanGain.java]: every place that set the hopper(Ballio) now is a runif that only runs when the override is off. There is another branch(the last one I made), which has accounts for all places the hopper is set, shooter included

**Review by c0d3rman — APPROVED** [review]
Still good by me.

**Pull request merged**